src/control/lib/hostlist/numericlist.go

```diff
@@ -24,19 +123,66 @@ func CreateNumericList(stringRanges string) (*HostList, error) {
 		}
 	}
 
-	return parseBracketedHostList(stringRanges, outerRangeSeparators,
+	hl, err := parseBracketedHostList(stringRanges, outerRangeSeparators,
 		rangeOperator, true)
+	if err != nil {
+		return nil, err
+	}
+	return &NumericList{hl: hl}, nil
+}
+
+// NumericSet is a special case of a NumericList that contains unique numeric
+// entries.
+type NumericSet struct {
+	NumericList
+}
+
+// Add adds the supplied numeric entry to the NumericSet.
+func (ns *NumericSet) Add(i uint) {
+	ns.NumericList.Add(i)
+	ns.Uniq()
+}
+
+// Delete removes the supplied numeric entry from the NumericSet. No-op if
+// the entry is not present.
+func (ns *NumericSet) Delete(i uint) {
+	ns.NumericList.Delete(i)
+	ns.Uniq()
```

Comment:
Should not be be needed

Reply:
The `HostList.Uniq()` method is responsible for doing any range consolidation and internal bookkeeping (e.g. updating count, etc). The `NumericSet` type is mostly just a convenience to handle calling `Uniq()` after any modification operation.

Reply:
Clarify, thanks to @mjmac explanation: missed that we want to keep a compact form.